lib/api/model/submessage.dart

```diff
@@ -313,3 +349,144 @@ class UnknownPollEventSubmessage extends PollEventSubmessage {
   @override
   Map<String, Object?> toJson() => json;
 }
+
+/// States of a poll Zulip widget.
+///
+/// See also:
+/// - https://zulip.com/help/create-a-poll
+/// - https://github.com/zulip/zulip/blob/304d948416465c1a085122af5d752f03d6797003/web/shared/src/poll_data.ts
+class Poll {
```

Comment:
I have been inclined to squash the `handleSubmessageEvent` commit with this commit since last update, because all the tests here would be moved to the later commit.

An alternative would be retrofitting the updated tests in the earlier commit with as little change as necessary (to make the diffs look better), then moving it. This way we have atomicity in both commits.

I think it makes sense to leave the isolated exercises of `_applyEvent` in the later commit that introduces submessage events, because that's when we have a public interface to access `_applyEvent`.

Reply:
I see, in the new revision this earlier commit adds tests in `test/api/model/submessage_test.dart`, and then the later commit moves them and adapts them to `test/model/message_test.dart`.

I don't really understand why the tests can't be in the latter form already in the earlier commit. The polls are there on the messages in the message store, right? So the tests that find them in the message store should work fine.

Reply:
If we really would have to move around and rewrite the tests like that, then it'd be better to squash the commits.

Reply:
Yeah, I think I overlooked the fact that those `MessageEvent`-based tests are not blocked by `SubmessageEvent`. My plan in an earlier version of this revision was to reintroduce all tests, including the `applyEvent` ones, in their rewritten form without event handling. I agree that leaving `submessage_test.dart` out from the very beginning is a better decision.